debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/OpenShiftUtils.java

```diff
@@ -260,6 +262,27 @@ public void waitForPods(String project, Map<String, String> labels) {
         }
     }
 
+    public void deletePodsOfDeployment(Deployment deployment) {
+        client.apps()
+                .deployments()
+                .inNamespace(deployment.getMetadata().getNamespace())
+                .withName(deployment.getMetadata().getName()).scale(0);
+        waitForPodsDeletion(deployment);
+    }
+
+    public void waitForPodsDeletion(Deployment deployment) {
+        String deploymentName = deployment.getMetadata().getName();
+        LOGGER.info("Waiting for pods to delete [" + deploymentName + "]");
+        Supplier<PodList> podListSupplier = () -> client.pods()
+                .inNamespace(deployment.getMetadata().getNamespace())
+                .withLabels(Map.of("deployment", deploymentName))
+                .list();
+        await().atMost(scaled(1), MINUTES)
+                .pollDelay(5, SECONDS)
+                .pollInterval(3, SECONDS)
```

Comment:
This wouldn't really work if the purpose was to "wait for pods to delete". Deployment acts as a pod controller. If you just delete pods it will automatically recreate them. Which means that in the time between pollIntervals the pods would very likely by recreated and thus this method would be unreliable...

`waitForDeploymentToScaleDown` or something similar is a more fitting name